pkg/volume/azure_dd/azure_dd.go

```diff
@@ -146,6 +148,45 @@ func (plugin *azureDataDiskPlugin) GetVolumeLimits() (map[string]int64, error) {
 		return nil, fmt.Errorf("Expected Azure cloudprovider, got %s", cloud.ProviderName())
 	}
 
+	instances, ok := cloud.Instances()
+	if !ok {
+		glog.Warningf("Failed to get instances from cloud provider")
+		return volumeLimits, nil
+	}
+
+	instanceType, err := instances.InstanceType(context.TODO(), plugin.host.GetNodeName())
+	if err != nil {
+		glog.Errorf("Failed to get instance type from Azure cloud provider")
+		return volumeLimits, nil
+	}
+
+	az, err := getCloud(plugin.host)
+	if err != nil {
+		glog.Errorf("failed to get azure cloud in GetVolumeLimits, plugin.host: %s", plugin.host.GetHostName())
+		return volumeLimits, nil
+	}
+
+	result, err := az.VirtualMachineSizesClient.List(context.TODO(), az.Location)
```

Comment:
Should we add a cache for this? the sizes won't change frequently

Reply:
added a cache in new commit